src/ciphersuite/mod.rs

```diff
@@ -462,10 +515,44 @@ impl AeadKey {
 }
 
 impl AeadNonce {
-    /// Build a new nonce for an AEAD from `Secret`.
-    pub(crate) fn from_secret(secret: Secret) -> Self {
+    /// Create an `AeadNonce` from a `Secret`. TODO: This function should
+    /// disappear when tackling issue #103.
+    pub fn from_secret(secret: Secret) -> Self {
+        let mut nonce = [0u8; NONCE_BYTES];
+        nonce.clone_from_slice(&secret.value);
```

Comment:
This panics if the lengths don't match. If there's any chance that this might happen we should check the length. But maybe not too important as this code should go away. But it's also relevant for the other uses of `clone_from_slice`.

Reply:
That's true, but I'm not sure what to do here. Evercrypt wants it to be a fixed array, so we run into this problem no matter what. I'm tempted to just delegate it to #228, where we have to do a bunch more length-checking in the context of `Secret`s anyway. What do you think?

Reply:
Sure, it's safe. Panicking is just not great unless we know that it really shouldn't happen.